doc/whats_new/v0.23.rst

```diff
@@ -187,6 +187,10 @@ Changelog
   `ValueError` for arguments `n_classes < 1` OR `length < 1`.
   :pr:`16006` by :user:`Rushabh Vasani <rushabh-v>`.
 
+- |API| The `StreamHandler` was removed from `sklearn.logger`. To continue to
+  receive logging messages from :mod:`sklearn.datasets`, a handler should
+  be added to `sklearn.logger`. :pr:`16451` by :user:`Christoph Deil <cdeil>`.
```

Comment:
I think the statement as-is in the changelog isn't 100% correct. In many cases the last resort handler is in place and will log messages by default to the console. If the user wants to change logging, they can add handlers to the sklear.logger, but e.g. also the root logger.

Maybe not try to explain how logging handlers work via the what's new entry, and to just explain why the handler was removed?

```suggestion
- |API| The `StreamHandler` was removed from `sklearn.logger` to avoid
  double logging of messages in common cases where a hander is attached
  to the root logger, and to follow the Python logging documentation
  recommendation for libraries to leave the log message handling to
  users and application code. :pr:`16451` by :user:`Christoph Deil <cdeil>`.
```

Reply:
> I think the statement as-is in the changelog isn't 100% correct. In many cases the last resort handler is in place and will log messages by default to the console. 

The last resort handler will log warnings to stderr. The `datasets` module logs with `info` and `debug` levels, and nothing is shown on the console.

Although, I am okay with updating the whats new to your suggestion.

Reply:
For example:

```py
from sklearn.datasets import fetch_kddcup99
fetch_kddcup99(data_home="data")
```

will print `Downloading https://ndownloader.figshare.com/files/5976042` on master, and nothing with this PR.

Reply:
I see. So [here](https://github.com/scikit-learn/scikit-learn/blob/1ba06518c5c4c7c1865110a8f34c4da64d8e478f/sklearn/datasets/_kddcup99.py#L45) the `sklearn.datasets._kddcup99` logger is created, it has the default level of warning and thus the info message isn't shown.

You could add a `logger.setLevel(logging.INFO)` there if you want it to appear by default for users.

My understanding is that messing with handlers in libraries is bad (causes confusion and double log messages for users), but messing with log levels is OK in libraries, to get the default behaviour you want.

Another option could be to use the `sklearn.logger` there to emit the message, but then I think you have to be very careful to avoid circular imports, probably is tricky.

Reply:
The lastresort handler will still filter everything out.

The only want to get the logging messages to appear is to change the logger to `logger.warning`.

Anyways, I am okay with removing the messages so we can be a better citizen of using logging.

Reply:
Hm that seems fine as info and debug seem to be the correct logging levels. We could set the level to 'warning' if we think that downloading the data is unexpect, but 'info' seems right and so it shouldn't be printed by default.